server/odc-common/src/test/java/com/oceanbase/odc/common/util/StringUtilsTest.java

```diff
@@ -398,4 +400,31 @@ public void startWithIgnoreSpaceAndNewLines_targetsNonStartingWithPrefixWithMaxM
         Assert.assertTrue(StringUtils.startsWithIgnoreSpaceAndNewLines(target, prefix, true, 2));
     }
 
+    @Test
+    public void escapeRegex_null_isNotPresent() {
+        Assert.assertFalse(StringUtils.escapeRegex(null).isPresent());
+    }
+
+    @Test
+    public void escapeRegex_stringContainsRegularChar_escapeSuccess() {
+        String regex = ".*+?^$[](){}|\\";
+        String expect = "\\.\\*\\+\\?\\^\\$\\[\\]\\(\\)\\{\\}\\|\\\\";
+        Optional<String> escapedRegex = StringUtils.escapeRegex(regex);
+        Assert.assertTrue(escapedRegex.isPresent());
+        Assert.assertEquals(expect, escapedRegex.get());
+    }
+
+    @Test
+    public void replaceFirst_plEditReplaceOriginalName_success() {
+        String pl =
+                "create function `FuN_test@111--#%&*()_+` (`p1` int(12)) returns int(11) begin declare v1 int; set v1 = p1 + 1; return v1; end;";
+        String originalName = "FuN_test@111--#%&*()_+";
+        String replaceName = "replace_name";
+        String expect =
+                "create function `replace_name` (`p1` int(12)) returns int(11) begin declare v1 int; set v1 = p1 + 1; return v1; end;";
+        Optional<String> escapedRegex = StringUtils.escapeRegex(originalName);
+        String actual = pl.replaceFirst(escapedRegex.get(), replaceName);
+        Assert.assertEquals(expect, actual);
```

Comment:
why the expect function name is "replace_name"?
the expect function name should be "Optional[FuN_test@111--#%&\*\(\)_\+]"？

Reply:
That's no problem.That is mainly to test the `repalceFirst` method. The first occurrence of the `originalName` in the `pl` below needs to be replaced
![image](https://github.com/user-attachments/assets/388c6533-20dc-4f91-9493-d61580c6284f)

Reply:
The test method name has changed to` replaceFirst_replaceOriginalNameContainingRegularCharsInPlEdit_success`